### PR TITLE
Formats: Add support for Fluent

### PR DESCRIPTION
**Pull request by str4d**
## Proposed changes

This PR adds basic support for Fluent files (issue #1926).

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
  - I haven't been able to get the tests to run to completion locally (or if they do, it will take a very long time and looks like it is looping from the log output).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information

This PR requires https://github.com/translate/translate/pull/4378. Both PRs require some testing by people who know how `translate-toolkit` and `weblate` work (i.e. not me!) to flush out any edge cases I missed.

### CODE REVIEW

**Comment by nijel** [comment]
The full testsuite takes very long to run, try running just a relevant subset, for example `./manage.py test weblate.formats`.

File: docs/formats.rst [hunk 1]
```diff
@@ -1485,6 +1487,39 @@ TBX is an XML format for the exchange of terminology data.
...
+Fluent is a monolingual text format that focuses on asymmetric localization: a
+simple string in one language can map to a complex multi-variant translation in
+another language.
+
++-------------------------------------------------------------------+
+| Typical Weblate :ref:`component`                                  |
++================================+==================================+
+| Filemask                       | ``i18n/*/component-name.flt``    |
++--------------------------------+----------------------------------+
+| Monolingual base language file | `Empty`                          |
```
Comment (nijel) [inline docs/formats.rst:1510]: I guess this should be set as this is monolingual format:
```suggestion
| Monolingual base language file | ``i18n/en/component-name.flt``   |
```

File: docs/formats.rst [hunk 2]
```diff
@@ -1485,6 +1487,39 @@ TBX is an XML format for the exchange of terminology data.
...
++--------------------------------+----------------------------------+
+| Template for new translations  | `Empty`                          |
++--------------------------------+----------------------------------+
+| File format                    | `Fluent file`                    |
++--------------------------------+----------------------------------+
+
+.. seealso::
+
+    `Project Fluent <https://www.projectfluent.org/>`_,
```
Comment (nijel) [inline docs/formats.rst:1519]:
```suggestion
    `Project Fluent <https://projectfluent.org/>`_,
```

File: requirements.txt [hunk 3]
```diff
@@ -12,6 +12,7 @@ django-redis>=4.11.0,<6.0.0
 Django[argon2]>=3.2,<3.3
 djangorestframework>=3.11.0,<3.13.0
 filelock>=3.0.0,<3.1.0
+fluent.syntax>=0.18.1,<0.19.0
```
Comment (nijel) [inline requirements.txt:15]: This should go to requirements-optional.txt, not everybody needs this.

File: weblate/formats/tests/test_formats.py [hunk 4]
```diff
@@ -1133,3 +1135,20 @@ class TBXFormatTest(AutoFormatTest):
     FIND_MATCH = "adresní řádek"
     NEW_UNIT_MATCH = b"<term>Source string</term>"
     EXPECTED_FLAGS = ""
+
+
+class FluentFormatTest(AutoFormatTest):
+    FORMAT = FluentFormat
+    FILE = TEST_FLUENT
+    BASE = ""
+    MIME = ""
```
Comment (nijel) [inline weblate/formats/tests/test_formats.py:1144]: It should have a MIME type, based on https://gitlab.gnome.org/GNOME/gtksourceview/-/merge_requests/87/diffs?commit_id=df7a2a1638f9661a91d4d27ae71b2ce506fd7bcb:
```suggestion
    MIME = "text/x-fluent"
```

**Review by nijel — COMMENTED** [review]
I have added some feedback. The tests on edge deps (which is using translate-toolkit from the master branch) should pass once https://github.com/translate/translate/pull/4378 is merged, and we can merge it then. It will produce a warning for people on older translate-toolkit, but that is not that big deal and I should be able to release translate-toolkit with Fluent support soon.
